### PR TITLE
Added typescript definition

### PR DESCRIPTION
**Pull request by AndrewLeedham**
## Description
Adds typescript definitions to the module.

## Related Issue

fixes: https://github.com/adobe/jwt-auth/issues/28

## Motivation and Context

Allows the module to be consumed via TypeScript.

## How Has This Been Tested?

Used `yarn link` locally and included it in a TypeScript project.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Type definitions for @adobe/jwt-auth 0.3
+// Project: https://github.com/adobe/jwt-auth#readme
+// Definitions by: Andrew Leedham <https://andrewleedham.me>
+
```
[the remark this answers is not in the document]
Comment (AndrewLeedham) [reply index.d.ts]: Sure, done.

**Comment by AndrewLeedham** [comment]
@ittaibaratz any other changes?

**Comment by ittaibaratz** [comment]
@macdonst can you review? I am not sure about the Travis changes as well as including contributor personal details in the code.

**Comment by AndrewLeedham** [comment]
> @macdonst can you review? I am not sure about the Travis changes as well as including contributor personal details in the code.

@ittaibaratz I can take the TS header out if you want that is just the standard recommended details from the template here: https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html

**Comment by ittaibaratz** [comment]
> > @macdonst can you review? I am not sure about the Travis changes as well as including contributor personal details in the code.
> 
> @ittaibaratz I can take the TS header out if you want that is just the standard recommended details from the template here: https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html

@AndrewLeedham it would be great if you can use our standard header without any personal details. Your details as a contributor are visible to everyone through Github itself.

Also, the Typescript addition looks good to me, however I am not sure about the Travis CI changes and its a bit challenging to test it - So if you want the Typescript addition added soon, I'd recommend you break this pull request into 2 pull requests as these are two separate issues.

**Comment by AndrewLeedham** [comment]
@ittaibaratz Ok I will refactor the comment header. I looped the travis changes in here because the tests will not pass for the PR otherwise.

**Comment by ittaibaratz** [comment]
> @ittaibaratz Ok I will refactor the comment header. I looped the travis changes in here because the tests will not pass for the PR otherwise.

That's ok, we can still merge the pull request if we are confident the tests are not failing because of the code change.

**Comment by AndrewLeedham** [comment]
Seems travis may have updated nvs or fixed the mismatch between linux and windows 🎉

**Pull request merged**